static-assets/components/cstudio-forms/controls/rte.js

```diff
@@ -279,248 +283,263 @@ CStudioAuthoring.Module.requireModule(
           this.rteId = rteId;
           this.rteWidth = '100%';
 
-          inputEl = this._renderInputMarkup(config, rteId);
-
-          // Getting properties from content-type
-          for (var i = 0; i < config.properties.length; i++) {
-            var prop = config.properties[i];
-
-            switch (prop.name) {
-              case 'imageManager':
-                this.imageManagerName = prop.value && prop.Value != '' ? prop.value : null;
-                break;
-              case 'videoManager':
-                this.videoManagerName = prop.value && prop.Value != '' ? prop.value : null;
-                break;
-              case 'audioManager':
-                this.audioManagerName = prop.value && prop.Value !== '' ? prop.value : null;
-                break;
-              case 'fileManager':
-                this.fileManagerName = prop.value && prop.Value != '' ? prop.value : null;
-                break;
-              case 'height':
-                this.rteHeight = prop.value === undefined || prop.value === '' ? 300 : parseInt(prop.value, 10);
-                break;
-              case 'autoGrow':
-                this.autoGrow = prop.value == 'false' ? false : true;
-                break;
-              case 'maxlength':
-                inputEl.maxlength = prop.value;
-                break;
-              case 'enableSpellCheck':
-                this.enableSpellCheck = !prop.value || prop.value === 'true';
-                break;
+          const { inputEl, onDemandEditorInitializationEl } = this._renderInputMarkup(config, rteId);
```

Comment:
Not ideal to change the return signature of this function. 
Could we: 
- use the same element for the preview (inputEl)
- create another function to do this work when the property is true?
- Set the new element to a `this` property and get it there from instead of from this return value. Looks like you're already setting this element to this somewhere else to `this.onDemandEditorInitializationEl` so perhaps set it at once on the renderInputMarkup?